schema/deploy/function_set_random_id.sql

```diff
@@ -0,0 +1,37 @@
+-- Deploy ggircs-portal:function_set_random_id to pg
+-- requires: schema_ggircs_portal
+
+begin;
+create extension if not exists pgcrypto;
+create function ggircs_portal.set_random_id()
+  returns trigger as $$
+
+declare
+  new_id varchar(1000);
+begin
+  while new_id in (select id from ggircs_portal.certification_url) or new_id is null loop
+    select gen_random_bytes(64) into new_id;
+  end loop;
```

Comment:
This loop avoids a [birthday attack](https://en.wikipedia.org/wiki/Birthday_attack) by detecting and avoiding hash collisions by re-running our CSPRNG function until there is no collision. @Krocodial let me know if you want to run through this together?

@dleard this is looking really good. Can we add a collision warning? Something like `raise warning 'ggics_portal.set_random_id() experienced hash collision on % – rerunning hash function', new_id;` This way, if we run out of address space in our hash column, or our CSPRNG fails or is compromised we'll have some logging to let us know what's going on.